eq-author/src/App/folder/Design/index.js

```diff
@@ -34,14 +37,34 @@ const StyledPanel = styled(Panel)`
   }
 `;
 
-const FolderDesignPage = ({ match }) => {
-  const { folderId } = match.params;
+const FolderDesignPage = ({ history, match }) => {
+  const { folderId, questionnaireId } = match.params;
 
   const { loading, error, data } = useQuery(GET_FOLDER_QUERY, {
     variables: { input: { folderId } },
   });
 
+  let sectionId, folderPosition, pages;
+  if (data) {
+    sectionId = data.folder.section.id;
+    folderPosition = data.folder.position;
+    pages = data.folder.pages;
+  }
```

Comment:
You can omit the conditional and move lined 49, 50 and 51 to after the (what is currently) line 97, the conditional to check if `data` is `null` or `undefined`.

`sectionId`, `folderPosition` and `pages` will then update in the same way you have now, but without checking to see if `data` is `null` or `undefined` twice.

Reply:
ahhh - of course - thanks Matt!